tests/test_sub_port_intf.py

```diff
@@ -851,23 +915,32 @@ def _test_sub_port_intf_removal(self, dvs, sub_port_intf_name, removal_seq_test=
         # Verify that sub port router interface entry is removed from ASIC_DB
         self.check_sub_port_intf_key_removal(self.asic_db, ASIC_RIF_TABLE, rif_oid)
 
-        # Verify physical port host interface vlan tag attribute
-        fv_dict = {
-            "SAI_HOSTIF_ATTR_VLAN_TAG": "SAI_HOSTIF_VLAN_TAG_STRIP",
-        }
-        for phy_port in phy_ports:
-            hostif_oid = dvs.asicdb.hostifnamemap[phy_port]
-            self.check_sub_port_intf_fvs(self.asic_db, ASIC_HOSTIF_TABLE, hostif_oid, fv_dict)
+        if removal_seq_test == True:
+            # Verify that parent port is removed from ASIC_DB
+            self.asic_db.wait_for_n_keys(asic_tbl_name, obj_cnt - 1)
+        else:
+            # Verify physical port host interface vlan tag attribute
+            fv_dict = {
+                "SAI_HOSTIF_ATTR_VLAN_TAG": "SAI_HOSTIF_VLAN_TAG_STRIP",
+            }
+            for phy_port in phy_ports:
+                hostif_oid = dvs.asicdb.hostifnamemap[phy_port]
+                self.check_sub_port_intf_fvs(self.asic_db, ASIC_HOSTIF_TABLE, hostif_oid, fv_dict)
 
         # Remove vrf if created
         if vrf_name:
             self.remove_vrf(vrf_name)
             self.asic_db.wait_for_n_keys(ASIC_VIRTUAL_ROUTER_TABLE, 1)
 
-        if parent_port.startswith(LAG_PREFIX):
-            # Remove lag members from lag parent port
-            self.remove_lag_members(parent_port, self.LAG_MEMBERS_UNDER_TEST)
-            self.asic_db.wait_for_n_keys(ASIC_LAG_MEMBER_TABLE, 0)
+        if parent_port.startswith(ETHERNET_PREFIX):
+            if removal_seq_test == True:
+                self.create_phy_port_appl_db(dvs, parent_port)
+                self.asic_db.wait_for_n_keys(asic_tbl_name, obj_cnt)
```

Comment:
Why is this needed?

Reply:
This is on the test cleanup path. It makes sure new physical port has been created in ASIC_DB before the next call to the test. There is pre-test check on the # of physical ports in ASIC_DB. Here, we make sure the ASIC_DB state is restored to before the test is conducted in terms of physical port count, and hope this may help increasing test stability for next test call.